sunpy/instr/goes.py

```diff
@@ -583,16 +605,490 @@ def _goes_get_chianti_em(longflux, temp, satellite=8, abundances="coronal",
+
+    force_download : (optional) bool
+        If True, the GOES radiative loss data file is downloaded.  It is
+        important to do this if a new version of the files has been
+        generated due to a new CHIANTI version being released or the
+        launch of new GOES satellites.
+        Default=False
+
+    download_dir : (optional) string
+        The directory to download the GOES radiative loss data file to.
+        Default=SunPy default download directory
+
+    Returns
+    -------
+    radlossrate : numpy ndarray, dtype=float, units=[erg]
+        Array containing radiative loss rates of the coronal plasma
+        corresponding to temperatures and emission measures in temp and
+        em arrays.
+
+    Notes
+    -----
+    This function calls a csv file containing a table of radiative loss
+    rate per unit emission measure at various temperatures.  The
+    appropriate values are then found via interpolation.  This table
+    was generated using CHIANTI atomic physics database employing the
+    methods of Cox & Tucker (1969).  Coronal abundances, a default
+    density of 10**10 cm**-3, and ionization equilibrium of
+    Mazzotta et al. (1998) were used.
+
+    References
+    ----------
+    .. [1] Cox, D. P., Tucker, W. H. 1969, ApJ, 157, 1157
+    .. [2] Mazzotta, P., Mazzitelli, G., Colafrancesco, S., & Vittorio, N.
+       1998, A&AS, 133, 339
+
+    Examples
+    --------
+    >>> temp = np.array([11.0, 11.0])
+    >>> em = np.array([4.0e+48, 4.0e+48])
+    >>> rad_loss = calc_rad_loss(temp, em)
+    >>> rad_loss["rad_loss_rate"]
+    array([  3.01851392e+26,   3.01851392e+26])
+    """
+    # Check inputs are correct
+    temp = temp.to(u.K)
+    em = em.to(1/u.cm**3)
+    if len(temp) != len(em):
+        raise ValueError("temp and em must all have same number of elements.")
+    # If force_download kwarg is True, or required data files cannot be
+    # found locally, download required data files.
+    check_download_file(FILE_RAD_COR, GOES_REMOTE_PATH, download_dir,
+                        replace=force_download)
+
+    # Initialize lists to hold model data of temperature - rad loss rate
+    # relationship read in from csv file
+    modeltemp = [] # modelled temperature is in log_10 sapce in units of MK
+    model_loss_rate = []
+
+    # Read data from csv file into lists, being sure to skip commented
+    # lines begining with "#"
+    with open(os.path.join(DATA_PATH, FILE_RAD_COR),
+              "r") as csvfile:
+        startline = csvfile.readlines()[7:]
+        csvreader = csv.reader(startline, delimiter=" ")
+        for row in csvreader:
+            modeltemp.append(float(row[0]))
+            model_loss_rate.append(float(row[1]))
+    modeltemp = np.asarray(modeltemp)
+    model_loss_rate = np.asarray(model_loss_rate)
+    # Ensure input values of flux ratio are within limits of model table
+    if np.min(temp.value) < np.min(modeltemp) or \
+        np.max(temp.value) > np.max(modeltemp):
```

Comment:
but... at the moment `temp` as `Quantity` is an `np.array` and so modeltemp, so you could do: 
```
temp.value.min() < modeltemp.min()
```
However, I think that it would be better to read the modeltemp from the file with the units (if available) and compare them directly.

Below, you could use `modeltemp.value` for the spline fit as with the `temp`.